bpf/lib/ipv4.h

```diff
@@ -44,8 +68,71 @@ static __always_inline bool ipv4_is_fragment(struct iphdr *ip4)
 	// +----+----+----+----------------------------------+
 	//
 	// If "More fragments" or the offset is nonzero, then this is an IP
-	// fragment. The evil bit must be set to 0 (RFC791, RFC3514).
-	return ip4->frag_off & bpf_htons(0xBFFF);
+	// fragment (RFC791).
+	return ip4->frag_off & bpf_htons(0x3FFF);
+}
+
+#if defined IPV4_FRAGMENTS
+static __always_inline bool ipv4_is_not_first_fragment(const struct iphdr *ip4)
+{
+	/* Ignore "More fragments" bit to catch all fragments but the first */
+	return ip4->frag_off & bpf_htons(0x1FFF);
+}
+
+static __always_inline int
+ipv4_frag_get_l4ports(const struct ipv4_frag_id *frag_id,
+		      struct ipv4_frag_l4ports *ports)
+{
+	struct ipv4_frag_l4ports *tmp;
+
+	tmp = map_lookup_elem(&IPV4_FRAG_DATAGRAMS_MAP, frag_id);
+	if (!tmp)
+		return DROP_FRAG_NOSUPPORT;
```

Comment:
Do you think we should use a different error code, eg `DROP_FRAG_NOT_FOUND` to differentiate this error vs. lack of support?

I can't decide whether this is a good idea or not. Technically we can determine whether this error code means that we don't support fragments or if it means that the fragment info was not found (eg out-of-order) because we will only log the message for this one either when fragment support is disabled for any fragment, or when fragment support is enabled with out-of-order / spurious random fragments not seen before. Would be cognitively easier to understand at a glance if at least the `cilium monitor` message made it clear which of these two cases are occurring though. Less digging through current state to understand.

Reply:
Makes sense, I thought of adding a new code but wasn't sure. As you say, it will probably make things more explicit, I'll add one.